artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/ActiveMQProtonRemotingConnection.java

```diff
@@ -119,10 +119,15 @@ public boolean isDestroyed() {
 
    @Override
    public void disconnect(boolean criticalError) {
-      ErrorCondition errorCondition = new ErrorCondition();
-      errorCondition.setCondition(AmqpSupport.CONNECTION_FORCED);
-      amqpConnection.close(errorCondition);
-      getTransportConnection().close();
+      amqpConnection.runLater(() -> {
+         ErrorCondition errorCondition = new ErrorCondition();
+         errorCondition.setCondition(AmqpSupport.CONNECTION_FORCED);
+         amqpConnection.close(errorCondition);
+         amqpConnection.flush();
+      });
+      amqpConnection.runLater(() -> {
```

Comment:
Probably it is a naive question, but why 2 separate runLater calls instead of one?

Reply:
@franz1981 During my work, the first one was runNow, the second Later. (I would accept better names BTW)

later I changed the first to later and forgot the other one :) 


fixing it on my next ammend